packages/runtime/test/integration/execute-transaction.ts

```diff
@@ -226,4 +251,264 @@ describe("Algorand Smart Contracts - Execute transaction", function () {
+		assert.equal(john.balance(), initialBalance);
+
+	});
+
+	it("Should do key registration, through execute transaction", () => {
+		const txSKParams: types.KeyRegistrationParam = {
+			type: types.TransactionType.KeyRegistration, // payment
+			sign: types.SignType.SecretKey,
+			fromAccount: john.account,
+			voteKey: vKey,
+			selectionKey: sKey,
+			voteFirst: 43,
+			voteLast: 1000,
+			voteKeyDilution: 5,
+			payFlags: { totalFee: 1000 },
+		};
+		const r = runtime.executeTx([txSKParams])[0] as BaseTxReceipt;
+		assert.isDefined(r);
+		assert.isDefined(r.txn);
+		assert.isDefined(r.txID);
+		syncAccounts();
+	});
+
+	it("Should modify asset, through execute transaction", () => {
+		setupAsset();
+		let modFields: types.AssetModFields = {
+			manager: elonMusk.address,
+			reserve: elonMusk.address,
+			clawback: alice.address,
+			freeze: alice.address,
+		};
+		const modifyParam: types.ModifyAssetParam = {
+			type: types.TransactionType.ModifyAsset,
+			sign: types.SignType.SecretKey,
+			fromAccount: john.account,
+			assetID: assetId,
+			fields: modFields,
+			payFlags: {},
+		};
+		runtime.executeTx([modifyParam]);
+		const res = runtime.getAssetDef(assetId);
+		assert.equal(res.manager, elonMusk.address);
+		assert.equal(res.reserve, elonMusk.address);
+		assert.equal(res.clawback, alice.address);
+		assert.equal(res.freeze, alice.address);
+	});
+
+	it("should freeze asset, through execute transaction", () => {
+		setupAsset();
+		runtime.optInToASA(assetId, alice.address, {});
+		const freezeParam: types.FreezeAssetParam = {
+			type: types.TransactionType.FreezeAsset,
+			sign: types.SignType.SecretKey,
+			fromAccount: john.account,
+			assetID: assetId,
+			freezeTarget: alice.address,
+			freezeState: true,
+			payFlags: {},
+		};
+		runtime.executeTx([freezeParam]);
+
+		const aliceAssetHolding = runtime.getAssetHolding(assetId, alice.address);
+		assert.equal(aliceAssetHolding["is-frozen"], true);
+	});
+
+	it("should revoke asset, through execute transaction", () => {
+		setupAsset();
+		runtime.optInToASA(assetId, alice.address, {});
+
+		const revokeParam: types.RevokeAssetParam = {
+			type: types.TransactionType.RevokeAsset,
+			sign: types.SignType.SecretKey,
+			fromAccount: john.account,
+			recipient: john.address,
+			assetID: assetId,
+			revocationTarget: alice.address,
+			amount: 0n,
+			payFlags: { totalFee: 1000 },
+		};
+		runtime.executeTx([revokeParam]);
+
+		syncAccounts();
+		assert.equal(john.balance(), initialBalance - 1000n);
+	});
+
+	it("Should destroy asset, through execute transaction", () => {
+		setupAsset();
+		const destroyParam: types.DestroyAssetParam = {
+			type: types.TransactionType.DestroyAsset,
+			sign: types.SignType.SecretKey,
+			fromAccount: john.account,
+			assetID: assetId,
+			payFlags: { totalFee: 1000 },
+		};
+
+		runtime.executeTx([destroyParam]);
+
+		syncAccounts();
+		assert.equal(john.balance(), initialBalance - 1000n);
+	});
```

Comment:
I recommend you should add check whether the asset is still exist or not

Reply:
ok. done